tests/Doctrine/Tests/DBAL/Platforms/MySqlPlatformTest.php

```diff
@@ -20,4 +26,32 @@ public function testHasCorrectDefaultTransactionIsolationLevel() : void
             $this->platform->getDefaultTransactionIsolationLevel()
         );
     }
+
+    public function testGetAlterTableSqlRemovesCharsetPlatformOptionWhenAlteringTypeFromStringTextToInteger(): void
+    {
+        $tableA     = new Table('tableA');
+        $textColumn = $tableA->addColumn('abc', Types::TEXT);
+        $textColumn->setPlatformOption('charset', 'UTF-8');
+        $stringColumn = $tableA->addColumn('def', Types::STRING);
+        $stringColumn->setPlatformOption('charset', 'SJIS');
+
+        $tableB      = clone $tableA;
+        $firstColumn = $tableB->getColumn('abc');
+        $firstColumn->setType(Type::getType(Types::INTEGER));
+        $secondColumn = $tableB->getColumn('def');
+        $secondColumn->setType(Type::getType(Types::INTEGER));
+
+        $comparator = new Comparator();
+        $tableDiff  = $comparator->diffTable($tableA, $tableB);
+
+        $this->platform->getAlterTableSQL($tableDiff);
+
+        self::assertGreaterThan(0, count($tableDiff->changedColumns));
+
+        foreach ($tableDiff->changedColumns as $columnDiff) {
+            $column = $columnDiff->column;
+
+            self::assertFalse($column->hasPlatformOption('charset'));
```

Comment:
Can you please assert the resulting SQL queries that are created by `getAlterTableSQL()`? You'll find similar tests in `AbstractMySQLPlatformTestCase`, so you don't need to check the columns definitions.

Reply:
@SenseException The purpose of this (unit) test is to test that the platform option is removed (which could be quite a bit more isolated using mocks I might add). Which is exactly what my change does under certain conditions.
Why are you asking me to implement a test to verify the complete behavior of the method of this derived platform class? I guess it is something the author of the method should have done.
The way I see it, my change is covered and it even increased the overall code coverage. If anything I'd say the test isn't testing the code in enough isolation and covering too much.
Anyway, like I explained before (and you did not acknowledge BTW), I'm done spending time on this.

Reply:
Thank you for your time you spent on this PR and your thoughts you put into this. I'll keep this PR open in case someone else wants to take over.